prov/gni/src/gnix_av.c

```diff
@@ -446,12 +446,12 @@ static int map_insert(struct gnix_fid_av *int_av, const void *addr,
 
 		((struct gnix_address *)fi_addr)[i] = temp.gnix_addr;
 		the_entry =  &blk->base[i];
-		memcpy(&the_entry->gnix_addr, &temp.gnix_addr,
-		       sizeof(struct gnix_address));
+		the_entry->gnix_addr.device_addr = temp.gnix_addr.device_addr;
+		the_entry->gnix_addr.cdm_id = temp.gnix_addr.cdm_id;
 		the_entry->name_type = temp.name_type;
 		the_entry->cm_nic_cdm_id = temp.cm_nic_cdm_id;
 		the_entry->cookie = temp.cookie;
-		memcpy(&key, &temp.gnix_addr, sizeof(gnix_ht_key_t));
+		key = *((gnix_ht_key_t *) &temp.gnix_addr);
```

Comment:
There is no reason to have a temporary variable for a gnix_ht_key unless the key that you intend to pass in is not 64 bits in length.

Inside the gnix_ht_insert function, the value of the key is stored in the entry object itself, so the passed-in key is only ever used as a temporary value store. If you do the following, you'd see the same behavior.

```C

ret = _gnix_ht_insert(int_av->map_ht, (gnix_ht_key_t *) &temp.gnix_addr, the_entry); 
```

See gnix_hashtable.c:660